engine/src/main/java/org/terasology/math/ChunkMath.java

```diff
@@ -0,0 +1,251 @@
+
+import org.terasology.math.geom.Vector3f;
+import org.terasology.world.chunks.ChunkConstants;
+
+/**
+ * Collection of math functions.
+ *
+ * @author Benjamin Glatzel <benjamin.glatzel@me.com>
+ */
+public final class ChunkMath {
+
+    private ChunkMath() {
+    }
+
+    /**
+     * Returns the chunk position of a given coordinate.
+     *
+     * @param x The X-coordinate of the block
+     * @return The X-coordinate of the chunk
+     */
+    public static int calcChunkPosX(int x, int chunkPowerX) {
+        return (x >> chunkPowerX);
+    }
+
+    /**
+     * Returns the chunk position of a given coordinate.
+     *
+     * @param y The Y-coordinate of the block
+     * @return The Y-coordinate of the chunk
+     */
+    public static int calcChunkPosY(int y, int chunkPowerY) {
+        return (y >> chunkPowerY);
+    }
+
+    /**
+     * Returns the chunk position of a given coordinate.
+     *
+     * @param z The Z-coordinate of the block
+     * @return The Z-coordinate of the chunk
+     */
+    public static int calcChunkPosZ(int z, int chunkPowerZ) {
+        return (z >> chunkPowerZ);
+    }
+
+    public static Vector3i calcChunkPos(Vector3i pos, Vector3i chunkPower) {
+        return calcChunkPos(pos.x, pos.y, pos.z, chunkPower);
+    }
+
+    public static Vector3i calcChunkPos(Vector3f pos) {
+        return calcChunkPos(new Vector3i(pos, 0.5f));
+    }
+
+    public static Vector3i calcChunkPos(Vector3i pos) {
+        return calcChunkPos(pos.x, pos.y, pos.z);
+    }
+
+    public static Vector3i calcChunkPos(int x, int y, int z) {
+        return calcChunkPos(x, y, z, ChunkConstants.CHUNK_POWER);
+    }
+
+    public static Vector3i[] calcChunkPos(Region3i region) {
+        return calcChunkPos(region, ChunkConstants.CHUNK_POWER);
+    }
+
+    public static Vector3i calcChunkPos(int x, int y, int z, Vector3i chunkPower) {
+        return new Vector3i(calcChunkPosX(x, chunkPower.x), calcChunkPosY(y, chunkPower.y), calcChunkPosZ(z, chunkPower.z));
+    }
+
+    public static Vector3i[] calcChunkPos(Region3i region, Vector3i chunkPower) {
+        int minX = calcChunkPosX(region.minX(), chunkPower.x);
+        int minY = calcChunkPosY(region.minY(), chunkPower.y);
+        int minZ = calcChunkPosZ(region.minZ(), chunkPower.z);
+
+        int maxX = calcChunkPosX(region.maxX(), chunkPower.x);
+        int maxY = calcChunkPosY(region.maxY(), chunkPower.y);
+        int maxZ = calcChunkPosZ(region.maxZ(), chunkPower.z);
+
+        int size = (maxX - minX + 1) * (maxY - minY + 1) * (maxZ - minZ + 1);
+
+        Vector3i[] result = new Vector3i[size];
+        int index = 0;
+        for (int x = minX; x <= maxX; x++) {
+            for (int y = minY; y <= maxY; y++) {
+                for (int z = minZ; z <= maxZ; z++) {
+                    result[index++] = new Vector3i(x, y, z);
+                }
+            }
+        }
+        return result;
+    }
+
+    /**
+     * Returns the internal position of a block within a chunk.
```

Comment:
Arguably "relative" to the chunk is confusing - can be read to mean a block adjacent to the chunk would have position (1,0,0). Relative to the "origin" of the chunk perhaps?